tests/chainer_tests/functions_tests/array_tests/test_spatial_transformer_grid.py

```diff
@@ -10,21 +10,45 @@
 from chainer.testing import attr
 
 
-@testing.parameterize(*testing.product({
-    'use_cudnn': ['always', 'never'],
-}))
+@testing.parameterize(*testing.product_dict(
+    [{'dtype': numpy.float16,
+      'forward_options': {'atol': 1e-4, 'rtol': 1e-3},
+      'backward_options': {'atol': 1e-2, 'rtol': 1e-1}},
+     {'dtype': numpy.float32,
+      'forward_options': {},
+      'backward_options': {}},
+     {'dtype': numpy.float64,
+      'forward_options': {},
+      'backward_options': {}},
+     ],
+    [{'use_cudnn': 'always'},
+     {'use_cudnn': 'never'}]
+))
 class TestSpatialTransformerGrid(unittest.TestCase):
 
     def setUp(self):
+        self._old_dtype = None
```

Comment:
How about using `using_config` by calling `__enter__` and `__exit__` directly?
```
    def setUp(self):
        self._config_user = chainer.using_config('dtype', self.dtype)
        self._config_user.__enter__()
        ...
    def tearDown(self):
        self._config_user.__exit__(None, None, None)
```

Reply:
(I added another comment below; that has higher priority so please look that first!)

Reply:
Fixed!